src/Type/Php/FilterVarDynamicReturnTypeExtension.php

```diff
@@ -231,7 +231,7 @@ private function getFlagsValue(Type $exprType): Type
 
 	private function canStringBeSanitized(Type $filterType, int $filterValue, ?Node\Arg $flagsArg, Scope $scope): bool
 	{
-		if (!$filterType instanceof StringType) {
+		if ($filterType->isSuperTypeOf(new StringType())->no()) {
```

Comment:
This doesn't seem right. For example what do you want to happen for these types?

* `mixed`
* `string|null`
* `string`
* `'1'`
* `int`

Reply:
I think you're missing where $filterType is coming from. It's the type returned from `getFilterTypeMap()`, which only ever returns `BooleanType`, `FloatType`, `IntType`, or `StringType`. So all I'm doing here is ensuring that we're only operating on filters that return strings. It's not verifying the type of the input itself, which is `in $inputType`. See where I'm calling it in [line 140](https://github.com/devbanana/phpstan-src/blob/filter_var-should-return-non-empty-string/src/Type/Php/FilterVarDynamicReturnTypeExtension.php#L140).

It's also why I felt OK using `instanceof StringType`.

Reply:
I'm confused. You're caling a method called `canStringBeSanitized` and if the input isn't a string, it returns `true`?

Reply:
Perhaps I should move that check to the condition on line 52, or else rename the method. if I return false, then it'll return non-empty-string if something like this is done:

```php
$str = 'foo';
filter_var($str, FILTER_VALIDATE_INT);
```

Since a string is being passed, but FILTER_VALIDATE_INT should return an int, not a string. So I only want to operate on filters that return strings. Let me move the check and see if that makes more sense.

Reply:
Yeah, I like this more, just make sure this case is also tested `filter_var('foo', FILTER_VALIDATE_INT);`

Reply:
Thanks, done.